apps/internal/oauth/ops/authority/authority.go

```diff
@@ -110,16 +110,20 @@ const (
 
 // AuthParams represents the parameters used for authorization for token acquisition.
 type AuthParams struct {
-	AuthorityInfo     Info
-	CorrelationID     string
-	Endpoints         Endpoints
-	ClientID          string
-	Redirecturi       string
-	HomeaccountID     string
-	Username          string
-	Password          string
-	Scopes            []string
-	AuthorizationType AuthorizeType
+	AuthorityInfo       Info
+	CorrelationID       string
+	Endpoints           Endpoints
+	ClientID            string
+	Redirecturi         string
+	HomeaccountID       string
+	Username            string
+	Password            string
+	Scopes              []string
+	AuthorizationType   AuthorizeType
+	State               string
+	CodeChallenge       string
+	CodeChallengeMethod string
+	Prompt              string
```

Comment:
Not sure if this should be a part of AuthParams, looks a bit odd conceptually, `AuthParams` are parameters used for authorization for token acquistion. These parameters are used for auth_code_acquistion. Having `code` in here would make more sense instead as that is something we use for token_acquistion.

Reply:
It's also used by `Client.AuthCodeURL()` to "creates a URL used to acquire an authorization code.".  So really at present it's used for both code and token acquisition.

I do see your point though that everything is lumped in here together making it a grab-bag of data with no clarity in what flows it's used.  If we want to refactor this I suggest we do it separately.  Thoughts?

Reply:
Okay, an issue tracking this would be good.

Reply:
Opened https://github.com/AzureAD/microsoft-authentication-library-for-go/issues/154